### PR TITLE
Update release checklist

### PR DESCRIPTION
**Pull request by ThomasLaPiana**
### Code Changes

* [ ] add functionality checks for ctl features
* [ ] add UI feature checks
* [ ] update ops feature checks

### Steps to Confirm

* [ ] _list any manual steps taken to confirm the changes_

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation Updated:
  - [ ] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  - [ ] documentation issue created (tag docs-team to complete issue separately)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`

### Description Of Changes

This PR updates the release checklist, so that we have a baseline of quality for every single release, including the initial release of unified-fides

### CODE REVIEW

**Comment by ThomasLaPiana** [comment]
@NevilleS @seanpreston @Kelsey-Ethyca would be happy to hear your input here as well (along with anyone else)

File: docs/fides/docs/development/release_checklist.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+- [ ] Run a standard evaluation
+- [ ] Scan the database
+
+## Admin UI
+
+- [ ] Every navigation button works
+
+## Privacy Center
+
+- [ ] Every navigation button works
```
Comment (seanpreston) [inline docs/fides/docs/development/release_checklist.md:29]: It'd be good to specifically have flows here — submit a DSR / consent request are probably the most useful

File: docs/fides/docs/development/release_checklist.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+## CLI
+
+- [ ] Run a local evaluation
+- [ ] Run a standard evaluation
+- [ ] Scan the database
+
+## Admin UI
+
+- [ ] Every navigation button works
```
Comment (seanpreston) [inline docs/fides/docs/development/release_checklist.md:25]: Then here, approve a DSR and check it succeeds would go a long way

Comment (ThomasLaPiana) [reply docs/fides/docs/development/release_checklist.md:25]: @seanpreston can you add these directly? I just want to make sure nothing gets lost in my translating your comments to items

**Review by conceptualshark — APPROVED** [review]

**Pull request merged**